### ISSUE TITLE
Seemingly Erroneous flagging of duplicate-id-aria

### ISSUE THREAD

**Issue by MattSchiller** [dequelabs/axe-core#2855]
I'm encountering an issue where it's reporting a11y violations, when I don't believe that it should. I'm rendering the same checkbox toggle element in two places, one whose parent/containing element is hidden with `display: none` on mobile widths, and one that's hidden in the same way on non-mobile widths. Both toggles are rendered from identical react components and have the same label.

Expectation: I would have expected the check to realize that only one of these items is actually accessible/viewable to my user and not thrown an error in this instance.

Actual: I'm getting the `duplicate-id-aria` error:
```
description: "Ensures every id attribute value used in ARIA and in labels is unique"
help: "IDs used in ARIA and labels must be unique"
helpUrl: "https://dequeuniversity.com/rules/axe/3.3/duplicate-id-aria?application=axeAPI"
id: "duplicate-id-aria"
impact: "critical"
```

Motivation: It's hindering my team's ability to re-use components across desktop and mobile patterns that use parallel render pathways and `display: none` to hide the one we don't need.

<pre><code>
axe-core version: 4.1.1
axe-webdriver, extension or other integration version: X.Y.Z

Browser and Assistive Technology versions
Chrome: 89.0.4389.90 

For Tooling issues:
- Node version: v12.16.0
- Platform: Mac
</code></pre>

**Comment by WilcoFiers** [comment]
@MattSchiller You should definitely change this. Using `display: none` doesn't remove an element from consideration of the accessible name computation. This is kind of a gotcha in ARIA (one of far too many, if you ask me).  I know this is intentional, but I can only speculate about the use case for it. I think it's just to give elements the best possible chance of actually ending up with an accessible name.

It's an easy fix, just take out one of the IDs. You shouldn't have them; they make screen reader behavior unpredictable, and you may end up with some of them reading out both labels, and others reading none of them.

**Comment by MattSchiller** [comment]
Gotcha. In parallel to seeing this error, I've refactored the offending page to be more truly responsive, but ah, I think I'm gonna have a harder time convincing my team to eschew this bifurcated render approach (however better it really is).

Thank you!
